rules/internal/utils/type_name.go

```diff
@@ -17,23 +17,17 @@ package utils
 import (
 	"strings"
 
-	dpb "github.com/golang/protobuf/protoc-gen-go/descriptor"
 	"github.com/jhump/protoreflect/desc"
 )
 
-type hasGetType interface {
-	GetType() dpb.FieldDescriptorProto_Type
-}
-
-// GetScalarTypeName returns the name of the type of the field, as a string.
-func GetScalarTypeName(t hasGetType) string {
-	return strings.ToLower(t.GetType().String()[len("TYPE_"):])
-}
-
-// GetMessageTypeName returns the name of the type of the field, as a string.
-func GetMessageTypeName(f *desc.FieldDescriptor) string {
-	if messageType := f.GetMessageType(); messageType != nil {
-		return messageType.GetFullyQualifiedName()
+// GetTypeName returns the name of the type of the field, as a string,
+// regardless
+func GetTypeName(f *desc.FieldDescriptor) string {
+	if m := f.GetMessageType(); m != nil {
+		return m.GetFullyQualifiedName()
```

Comment:
Are you sure to return the fully qualified name?
For example:
```
package test;
message Book {
  Author author = 1;
}
message Author{}
```
Will it return "test.Author" if we get the type name of `author`? But we really want "Author", right?

Reply:
I always want the fully qualified name in this case.